src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4822,6 +4822,183 @@ .ForwardHelpCategory Cmdlet
+
+    [string] $sudoArgs = Get-SudoArgs -AllArgs $AllArgs
+
+    <# tokenize the remaining arguments to identify the type of expression; native or powershell #>
+    $tokens = [System.Management.Automation.PSParser]::Tokenize($AllArgs.ToArray(), [ref] $null)
+    foreach ($token in $Tokens)
+    {
+        if ([string]::IsNullOrEmpty($token.Content) -or $script:IgnoredTokens.Contains($token.Type))
+        {
+            continue
+        }
+
+        if ($token.Type -eq 'Operator')
+        {
+            if ($AmbiguousOperators.Contains($token.Content))
+            {
+                continue
+            }
+            $usePowerShell = $true
+            break
+        }
+
+        if ($token.Type -eq 'Command')
+        {
+            $command = $token.Content
+            $currentCommand = Get-Command -Name $command -ErrorAction Ignore
+
+            if ($currentCommand -eq  $null)
+            {
+                break
+            }
+
+            if ($currentCommand.CommandType -eq 'Alias')
+            {
+                $aliasCommand = Get-Command -Name $currentCommand.Name -ErrorAction Ignore
+                if ($aliasCommand -ne $null)
+                {
+                    $currentCommand = Get-Command -Name $aliasCommand.ResolvedCommand -ErrorAction Ignore
+                }
+            }
+            if ($currentCommand.CommandType -ne 'Application')
+            {
+                $usePowerShell = $true
+                break
+            }
+            continue
+        }
+        $usePowerShell = $true
+        break
+    }
+
+    $sb = [System.Text.StringBuilder]::new()
+
+    [string] $commandArgs = $null
+    if ($AllArgs.Count -gt 0)
+    {
+        foreach ($arg in $AllArgs)
+        {
+            if ($sb.Length -gt 0)
+            {
+                $null = $sb.Append(' ')
+            }
+            $null = $sb.Append($arg)
+        }
+        $commandArgs = $sb.ToString()
+        $null = $sb.Clear()
+    }
+
+    $null = $sb.Append($SudoCmd)
+    if (-not [string]::IsNullOrEmpty($sudoArgs))
+    {
+        $null = $sb.Append(' ')
+        $null = $sb.Append($sudoArgs.ToString())
+    }
+
+    if (-not [string]::IsNullOrEmpty($commandArgs))
+    {
+        $null = $sb.Append(' ')
+        if (-not $usePowerShell)
+        {
+            $null = $sb.Append($commandArgs)
+        }
+        else
+        {
+            $bytes = [System.Text.Encoding]::Unicode.GetBytes($commandArgs)
+            $arguments = [Convert]::ToBase64String($bytes)
+            $null = $sb.AppendFormat([System.Globalization.CultureInfo]::InvariantCulture, 'powershell -encodedCommand {0}', $arguments)
+        }
+    }
+    return  [ScriptBlock]::Create($sb.ToString())
+}
+
+[string] $sudoCmd = $null
+$sudoCmds = Get-Command -Name 'sudo' -CommandType Application -ErrorAction Ignore
+if ($sudoCmds -eq $null)
+{
+    throw 'Could not resolve the sudo command'
+}
+
+if ($sudoCmds.GetType().IsArray)
+{
+    $sudoCmd = $sudoCmds[0].Path
```

Comment:
This is more complicated than necessary. You can force the result to be an array with:

```powershell
$sudoCmds = @(Get-Command ...)
```
Or you can force it to be a singleton with:
```powershell
$sudoCmds = Get-Command ... | Select -First 1
```

I think I prefer the later.